### PR TITLE
Task/rdmp 168 aws integration

### PR DESCRIPTION
**Pull request by JFriel**
## Proposed Change

Release engine block that writes releases to an S3 bucket
## Type of change

What types of changes does your code introduce? Tick all that apply.

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New Feature (non-breaking change which adds functionality)
-   [ ] Breaking Change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation-Only Update
-   [ ] Other  (if none of the other choices apply)

## Checklist

By opening this PR, I confirm that I have:

-   [x] Ensured that the PR branch is in sync with the target branch (i.e. it is automatically merge-able)
-   [x] Created or updated any tests if relevant
-   [x] Have validated this change against the [Test Plan](https://github.com/HicServices/RDMP/blob/develop/Documentation/CodeTutorials/TestPlan.md)
-   [x] Requested a review by one of the repository maintainers
-   [x] Have written new documentation or updated existing documentation to detail any new or updated functionality and how to use it
-   [x] Have added an entry into the changelog

### CODE REVIEW

File: Documentation/CodeTutorials/Packages.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-
+ataf
```
Comment (JFriel) [inline Documentation/CodeTutorials/Packages.md:1]: ?

File: Rdmp.Core/DataExport/DataRelease/AWSS3BucketReleaseDestination.cs [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+
+            listener.OnNotify(this, new NotifyEventArgs(ProgressEventType.Information,
+                $"Deleted {recordsDeleted} old CumulativeExtractionResults (That were not included in the final Patch you are preparing)"));
+
+        }
+        _engine = new AWSReleaseEngine(_project, ReleaseSettings, _s3Helper, _bucket, BucketFolder, listener, releaseAudit);
+        _engine.DoRelease(_releaseData.ConfigurationsForRelease, _releaseData.EnvironmentPotentials,
+            _releaseData.ReleaseState == ReleaseState.DoingPatch);
+        _configurationReleased = _engine.ConfigurationsReleased;
+        return null;
```
Comment (JFriel) [inline Rdmp.Core/DataExport/DataRelease/AWSS3BucketReleaseDestination.cs:263]: return release audit?

File: Rdmp.Core/ReusableLibraryCode/AWS/AWSS3.cs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        }
+        return bucket;
+    }
+
+    public static string KeyGenerator(string path, string file)
+    {
+        return Path.Join(path, file).Replace("\\", "/");
+    }
+
+    public async Task<bool> DoesObjectExists(string Key, string bucketName)
```
Comment (JFriel) [inline Rdmp.Core/ReusableLibraryCode/AWS/AWSS3.cs:76]: bad english

File: Rdmp.Core/ReusableLibraryCode/AWS/AWSS3.cs [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine(ex.Message);
+        }
+        return (response != null && response.S3Objects != null && response.S3Objects.Count > 0 && response.S3Objects.Any(o => o.Key == Key));
+    }
+
+
+    public bool ObjectExists(string fileKey, string bucketName)
```
Comment (JFriel) [inline Rdmp.Core/ReusableLibraryCode/AWS/AWSS3.cs:98]: do  we need both exist functions?

**Review by rdteviotdale — APPROVED** [review]

**Pull request merged**